cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -26,19 +26,584 @@ Cask Data Application Platform Release Notes
+  ``<base-url>/namespaces/<namespace-id>/data/explore/queries/count``.
+  
+- `CDAP-3516 <https://issues.cask.co/browse/CDAP-3516>`__ -
+  Added a service to manage the metadata of CDAP entities.
+  
+- `CDAP-3517 <https://issues.cask.co/browse/CDAP-3517>`__ -
+  Added data stores to manage business and lineage metadata for CDAP entities.
+  
+- `CDAP-3518 <https://issues.cask.co/browse/CDAP-3518>`__ -
+  Added the publishing of metadata change notifications to Apache Kafka.
+  
+- `CDAP-3519 <https://issues.cask.co/browse/CDAP-3519>`__ -
+  Added the ability to compute lineage of a CDAP dataset or stream in a given time window.
+  
+- `CDAP-3520 <https://issues.cask.co/browse/CDAP-3520>`__ -
+  Added RESTful APIs for adding/retrieving/deleting of metadata for apps/programs/datasets/streams.
+  
+- `CDAP-3521 <https://issues.cask.co/browse/CDAP-3521>`__ -
+  Added the ability to record a dataset or stream access by a CDAP program.
+  
+- `CDAP-3522 <https://issues.cask.co/browse/CDAP-3522>`__ -
+  Added the capability to search CDAP entities based on their metadata.
+  
+- `CDAP-3523 <https://issues.cask.co/browse/CDAP-3523>`__ -
+  Added RESTful APIs for searching CDAP entities based on business metadata.
+  
+- `CDAP-3549 <https://issues.cask.co/browse/CDAP-3549>`__ -
+  Added SSH port forwarding to the CDAP virtual machine.
+  
+- `CDAP-3563 <https://issues.cask.co/browse/CDAP-3563>`__ -
+  Removed faulty and unused metrics around CDAP file resource usage.
+  
+- `CDAP-3590 <https://issues.cask.co/browse/CDAP-3590>`__ -
+  Added the ability to write to multiple sinks in ETL realtime and batch applications.
+  
+- `CDAP-3626 <https://issues.cask.co/browse/CDAP-3626>`__ -
+  For the CSV and TSV stream formats, a "mapping" setting can now be specified, mapping
+  stream event columns to schema columns.
+  
+- `CDAP-3693 <https://issues.cask.co/browse/CDAP-3693>`__ -
+  Added support for CDAP to work with HDP 2.3.
+  
+- `CDAP-3787 <https://issues.cask.co/browse/CDAP-3787>`__ -
+  Provide support for writing to Elasticsearch from batch ETL applications.
+
+Improvements
+------------
+
+- `CDAP-2514 <https://issues.cask.co/browse/CDAP-2514>`__ -
+  Added an interface to load a file into a stream from the CDAP-UI.
+  
+- `CDAP-2592 <https://issues.cask.co/browse/CDAP-2592>`__ -
+  Updated the version of Angular used in the CDAP-UI to 1.4.
+
+- `CDAP-2723 <https://issues.cask.co/browse/CDAP-2723>`__ -
+  In the CDAP-UI, simplified the Projection properties shown.
+  
+- `CDAP-2809 <https://issues.cask.co/browse/CDAP-2809>`__ -
+  The CDAP-UI "Errors" pop-up in the main screen now displays the time and date for each
+  error.
+    
+- `CDAP-2872 <https://issues.cask.co/browse/CDAP-2872>`__ -
+  Updated the Cloudera Manager CSD to use support for logback.
+  
+- `CDAP-2917 <https://issues.cask.co/browse/CDAP-2917>`__ -
+  Resolved duplicated constants, ``Id.Namespace.DEFAULT`` and ``Constants.DEFAULT_NAMESPACE_ID``.
```

Comment:
this is just code improvement. It ended up in the Release Notes fields by mistake (by yours truly). I've removed the release notes field from the jira. Lets remove it from here.